### PR TITLE
Support Timestamp type for valueTypeEncoder

### PR DESCRIPTION
**Pull request by hanahmily**
I'm working on the integration between Apache Skywalking and istio, following [Mixer-Out-Of-Process-Adapter-Walkthrough](https://github.com/istio/istio/wiki/Mixer-Out-Of-Process-Adapter-Walkthrough).

But the mixer currently can't encode TIMESTAMP type which is be used by `request.time`, `response.time` etc. So I add a new switch-case branch in `valueTypeEncoder.go` to support it.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@mandarjog @ZackButcher This is the issue we have found, when we are trying to use out-of-processor adaptor for SkyWalking. Relate to #7683

Further, more data type should be supported, we just added the `Timestamp` which skywalking needed. Such as **duration** type may be an issue too.

File: mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ import (
 
 	"github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
 
+	"time"
```
Comment (ZackButcher) [inline mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go]: nit: imports should be in three blocks: standard library, third party, first party. Move time up to the top block and gogo proto types up beside the gogo descriptor import

File: mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go [hunk 2]
```diff
@@ -92,6 +96,16 @@ func valueTypeEncoderBuilder(_ *descriptor.DescriptorProto, fd *descriptor.Field
...
 					return nil
 				}
+			case v1beta1.TIMESTAMP:
+				enc = func(bag attribute.Bag, vVal *v1beta1.Value) error {
+					ev, er := expr.Evaluate(bag)
+					if er != nil {
+						return er
+					}
+					ts := ev.(time.Time)
+					vVal.Value = &v1beta1.Value_TimestampValue{&v1beta1.TimeStamp{&types.Timestamp{ts.Unix(), int32(ts.Nanosecond())}}}
```
Comment (ZackButcher) [inline mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go]: Use https://godoc.org/github.com/gogo/protobuf/types#TimestampProto to do this instead.

File: mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go (hunk 1 above)
Comment (hanahmily) [reply mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go]: done.

File: mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go (hunk 2 above)
Comment (hanahmily) [reply mixer/pkg/protobuf/yaml/dynamic/valueTypeEncoder.go]: done.

**Comment by mandarjog** [comment]
Thanks for doing this. Like you said several others are missing. Partially addresses https://github.com/istio/istio/issues/7645

**Comment by mandarjog** [comment]
Please check build failures.
Code looks good.
/lgtm

**Comment by wu-sheng** [comment]
@mandarjog Look like the `build` passed. The fails tasks look like not related to our codes. At the same time, master codebases have build failures too. https://circleci.com/gh/istio/istio .

@mandarjog @ZackButcher Could you help us with how to fix this?

**Comment by mandarjog** [comment]
`TestAgent - pilot` is always failing here ...

**Comment by geeknoid** [comment]
I'll force merge then.

**Comment by mandarjog** [comment]
I would much rather chase the person whose test is failing :-)  ...
let's do that still...

**Pull request merged**